src/group/libp2p_group_relcast_server.erl

```diff
@@ -182,15 +178,15 @@ handle_cast({send_ready, _Target, Index, Ready}, State0=#state{self_index=_SelfI
         false ->
             case Ready of
                 true ->
-                    {noreply, dispatch_next_messages(ready_worker(Index, undefined, State))};
+                    {noreply, dispatch_next_messages(ready_worker(Index, true, State))};
                 false ->
-                    {noreply, ready_worker(Index, false, State)}
+                    {noreply, dispatch_next_messages(ready_worker(Index, false, State))}
```

Comment:
Collapse the case to just `{noreply, dispatch_next_message(ready_worker(Index, Ready, Sate)}`?